src/GafferScene/LightToCamera.cpp

```diff
@@ -0,0 +1,312 @@
+			it != attributes->members().end(); it++ )
+		{
+			const std::string &attributeName = it->first.string();
+			if( !( boost::ends_with( attributeName, ":light" ) || attributeName == "light" ) )
+			{
+				continue;
+			}
+
+			const IECore::ObjectVector *shaderVector = IECore::runTimeCast<const IECore::ObjectVector>( it->second.get() );
+			if( !shaderVector || shaderVector->members().empty() )
+			{
+				continue;
+			}
+
+			IECore::InternedString shaderName;
+			const IECore::Shader *shader = IECore::runTimeCast<const IECore::Shader>( shaderVector->members().back().get() );
+			if( !shader )
+			{
+				continue;
+			}
+
+
+			std::string metadataTarget = attributeName + ":" + shader->getName();
+			const IECore::CompoundData *shaderParameters = shader->parametersData();
+			ConstStringDataPtr type = Metadata::value<StringData>( metadataTarget, "type" );
+
+			if( !type || !shaderParameters )
+			{
+				continue;
+			}
+
+			IECore::CameraPtr result = new IECore::Camera();
+
+
+			float screenWindowScale = 1.0f;
+			if( type->readable() == "distant" )
+			{
+			    const float locatorScale = parameter<float>( metadataTarget, shaderParameters, "locatorScaleParameter", 1 );
+				
+				result->parameters()["projection"] = new StringData( "orthographic" );
+				result->parameters()["clippingPlanes"] = new V2fData( V2f( -100000, 100000 ) );
+				screenWindowScale = locatorScale;
+			}
+			else if( type->readable() == "spot" )
+			{
+
+				/// \todo - this logic should look much cleaner once we have LightAccessor
+				float coneAngle = parameter<float>( metadataTarget, shaderParameters, "coneAngleParameter", 0.0f );
+				float penumbraAngle = parameter<float>( metadataTarget, shaderParameters, "penumbraAngleParameter", 0.0f );
+
+				if( ConstStringDataPtr angleUnit = Metadata::value<StringData>( metadataTarget, "angleUnit" ) )
+				{
+					if( angleUnit->readable() == "radians" )
+					{
+						coneAngle *= 180.0 / M_PI;
+						penumbraAngle *= 180 / M_PI;
+					}
+				}
+
+				const std::string *penumbraType = NULL;
+				ConstStringDataPtr penumbraTypeData = Metadata::value<StringData>( metadataTarget, "penumbraType" );
+				if( penumbraTypeData )
+				{
+					penumbraType = &penumbraTypeData->readable();
+				}
+
+				float outerAngle = coneAngle;
+				if( penumbraType && *penumbraType == "outset" )
+				{
+					outerAngle = coneAngle + 2.0f * penumbraAngle;
+				}
+				outerAngle = std::max( 0.25f, outerAngle );
+
+
+				if( parameter<bool>( metadataTarget, shaderParameters, "lensRadiusEnableParameter", true ) )
+				{
+					float lensRadius = parameter<float>( metadataTarget, shaderParameters, "lensRadiusParameter", 0.0f );
+					float focalPointOffset = lensRadius / tan( 0.5f * outerAngle * M_PI / 180.0f );
+
+					M44f transformOffset;
+					transformOffset.setTranslation( V3f( 0.0f, 0.0f, focalPointOffset ) );
+					result->parameters()["__projectionTransformOffset"] = new M44fData( transformOffset );
```

Comment:
I feel bad about repeating all computation twice, but I guess it shouldn't be a bottleneck.  I think I ended up going this way partly because I couldn't think of a clean way to factor it out - most of the logic is going to have to be repeated.  Maybe just have:
void convertLightToCamera( IECore::ConstCompoundObjectPtr attributes, int resolution, IECore::CameraPtr &outCamera, M44f &projectionTransformOffset ),
and always ignore one of the two outputs?